acme/api/order.go

```diff
@@ -0,0 +1,186 @@
+package api
+
+import (
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/go-chi/chi"
+	"github.com/pkg/errors"
+	"github.com/smallstep/certificates/acme"
+	"github.com/smallstep/certificates/api"
+	"github.com/smallstep/certificates/logging"
+)
+
+// NewOrderRequest represents the body for a NewOrder request.
+type NewOrderRequest struct {
+	Identifiers []acme.Identifier `json:"identifiers"`
+	NotBefore   time.Time         `json:"notBefore"`
+	NotAfter    time.Time         `json:"notAfter"`
+}
+
+// Validate validates a new-order request body.
+func (n *NewOrderRequest) Validate() error {
+	if len(n.Identifiers) == 0 {
+		return acme.MalformedErr(errors.Errorf("identifiers list cannot be empty"))
+	}
+	for _, id := range n.Identifiers {
+		if id.Type != "dns" {
+			return acme.MalformedErr(errors.Errorf("identifier type unsupported: %s", id.Type))
+		}
+	}
+	now := time.Now().UTC()
+	if n.NotBefore.IsZero() {
+		n.NotBefore = now
+	}
+	if n.NotAfter.IsZero() {
+		n.NotAfter = now.Add(time.Hour * 24)
```

Comment:
Is 24h in the spec? Is this a good enough value or a configuration value like `defaultTLSCertDuration`?

Reply:
The spec doesn't mention a default duration. This is just to match what we're already doing. I can open up the question to the group to see if we think acme provisioners should have a longer default duration. I think this will be a bit cleaner once the acme provisioner is implemented, because then the defaults will be in the provisioner.